api/server/server.go

```diff
@@ -260,52 +261,46 @@ func getContainersExport(eng *engine.Engine, version version.Version, w http.Res
 }
 
 func getImagesJSON(eng *engine.Engine, version version.Version, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
-	if err := parseForm(r); err != nil {
+	var err error
+
+	if err = parseForm(r); err != nil {
 		return err
 	}
 
-	var (
-		err  error
-		outs *engine.Table
-		job  = eng.Job("images")
-	)
-
-	job.Setenv("filters", r.Form.Get("filters"))
-	// FIXME this parameter could just be a match filter
-	job.Setenv("filter", r.Form.Get("filter"))
-	job.Setenv("all", r.Form.Get("all"))
+	imagesConfig := graph.ImagesConfig{
+		Filters: r.Form.Get("filters"),
+		// FIXME this parameter could just be a match filter
+		Filter: r.Form.Get("filter"),
+		All:    r.Form.Get("all") == "1",
+	}
 
-	if version.GreaterThanOrEqualTo("1.7") {
-		streamJSON(job, w, false)
-	} else if outs, err = job.Stdout.AddListTable(); err != nil {
+	images, err := getDaemon(eng).Repositories().CmdImages(&imagesConfig)
```

Comment:
We had discussion in IRC about that things. That api.types will be versioned and all. And internal datastructures won't be serialized to user. At least that I understood from @crosbymichael speech.